build/test_update_ext_version.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import json
+import pathlib
+import pytest
+
+from update_ext_version import main as update_version, parse_version
+
+
+@pytest.fixture
+def directory(tmpdir):
+    """Fixture to create a temp directory wrapped in a pathlib.Path object."""
+    return pathlib.Path(tmpdir)
+
+
+@pytest.mark.parametrize(
+    "version, args", [("1.1.0-rc", ["--release"]), ("1.0.0-rc", [])]
+)
+def test_minor_version(directory, version, args):
```

Comment:
This name is misleading since you're not testing the minor version, you're testing if a suffix without a version number fails.

```suggestion
def test_suffix_missing_version(directory, version, args):
```

Reply:
@brettcannon This is testing the odd/even thing. where if t is release then the minor has to be even. else it has to be odd.